nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-server/src/test/java/org/apache/nifi/distributed/cache/server/map/TestMapCacheClient.java

```diff
@@ -0,0 +1,146 @@
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class TestMapCacheClient {
+    private int port;
+    private EventServer server;
+    private final Serializer<String> stringSerializer = new StringSerializer();
+
+    @BeforeEach
+    public void setRunner() {
+        final TestRunner runner = TestRunners.newTestRunner(Mockito.mock(Processor.class));
+        port = NetworkUtils.getAvailableTcpPort();
+
+        final NettyEventServerFactory serverFactory = new NettyEventServerFactory(
+                InetAddress.getLoopbackAddress(), port, TransportProtocol.TCP);
+        serverFactory.setShutdownQuietPeriod(ShutdownQuietPeriod.QUICK.getDuration());
+        serverFactory.setShutdownTimeout(ShutdownQuietPeriod.QUICK.getDuration());
+        final ComponentLog log = runner.getLogger();
+        final VersionNegotiator versionNegotiator = new StandardVersionNegotiator(
+                ProtocolVersion.V3.value(), ProtocolVersion.V2.value(), ProtocolVersion.V1.value());
+
+        final MapCache mapCache = new SimpleMapCache("serviceIdentifier", 64, EvictionPolicy.FIFO) {
+            // override a service function that will cause client network timeout to fire
+            @Override
+            public MapPutResult put(ByteBuffer key, ByteBuffer value) throws IOException {
+                try {
+                    Thread.sleep(3000L);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+                return super.put(key, value);
+            }
+        };
+
+        serverFactory.setHandlerSupplier(() -> Arrays.asList(
+                new CacheVersionResponseEncoder(),
+                new CacheOperationResultEncoder(),
+                new MapRemoveResponseEncoder(),
+                new MapSizeResponseEncoder(),
+                new MapValueResponseEncoder(),
+                new MapCacheRequestDecoder(log, 64, MapOperation.values()),
+                new MapCacheRequestHandler(log, mapCache),
+                new CacheVersionRequestHandler(log, versionNegotiator)
+        ));
+        server = serverFactory.getEventServer();
+    }
+
+    @AfterEach
+    public void shutdownServer() {
+        server.shutdown();
+    }
+
+    /**
+     * Service will hold request long enough for client timeout to be triggered, thus causing the request to fail.
+     */
+    @Test
+    public void testClientTimeoutOnServerNetworkFailure() throws InitializationException, IOException {
+        final DistributedMapCacheClientService client = new DistributedMapCacheClientService();
+        final MockControllerServiceInitializationContext clientInitContext1 =
+                new MockControllerServiceInitializationContext(client, "client");
+        client.initialize(clientInitContext1);
+
+        final Map<PropertyDescriptor, String> clientProperties = new HashMap<>();
+        clientProperties.put(DistributedMapCacheClientService.HOSTNAME, "localhost");
```

Comment:
Recommend changing `localhost` to `127.0.0.1` to avoid unnecessary name resolution:
```suggestion
        clientProperties.put(DistributedMapCacheClientService.HOSTNAME, "127.0.0.1");
```